### DISCUSSION TITLE
STATUS6 doesn't show MQTT SSL state

### DISCUSSION THREAD

**Discussion by jeroenst** [arendst/Tasmota#12133] [Ideas]
### PROBLEM DESCRIPTION
_A clear and concise description of what the problem is._

When I request status info the state of using ssl/tls is missing from STATUS6

### REQUESTED INFORMATION
_Make sure your have performed every step and checked the applicable boxes before submitting your issue. Thank you!_

- [x] Read the [Contributing Guide and Policy](https://github.com/arendst/Tasmota/blob/development/CONTRIBUTING.md) and [the Code of Conduct](https://github.com/arendst/Tasmota/blob/development/CODE_OF_CONDUCT.md)
- [x] Searched the problem in [issues](https://github.com/arendst/Tasmota/issues)
- [x] Searched the problem in [discussions](https://github.com/arendst/Tasmota/discussions)
- [x] Searched the problem in the [docs](https://tasmota.github.io/docs/FAQ)
- [x] Searched the problem in the [chat](https://discord.gg/Ks2Kzd4)
- [x] Device used (e.g., Sonoff Basic): Sonoff zbbridge
- [x] Tasmota binary firmware version number used: 9.4.0.4
  - [x] Pre-compiled
  - [ ] Self-compiled
- [x] Flashing tools used: webpage
- [x] Provide the output of command: `Backlog Template; Module; GPIO 255`:
```lua
  Configuration output here:
19:53:25.096 SRC: Backlog
19:53:25.098 CMD: Group 0, Index 1, Command "GPIO", Data "255"
19:53:25.104 MQT: home/TASMOTA_ZIGBEE/stat/RESULT = {"GPIO0":{"320":"Led_i1"},"GPIO1":{"3552":"Zigbee Tx"},"GPIO2":{"0":"None"},"GPIO3":{"3584":"Zigbee Rx"},"GPIO4":{"5312":"Zigbee Rst1"},"GPIO5":{"0":"None"},"GPIO9":{"0":"None"},"GPIO10":{"0":"None"},"GPIO12":{"640":"I2C SDA"},"GPIO13":{"544":"LedLink"},"GPIO14":{"608":"I2C SCL"},"GPIO15":{"0":"None"},"GPIO16":{"32":"Button1"},"GPIO17":{"0":"None"}}

19:59:04.547 CMD: module
19:59:04.549 SRC: WebConsole from 172.30.30.100
19:59:04.550 CMD: Group 0, Index 1, Command "MODULE", Data ""
19:59:04.554 MQT: home/TASMOTA_ZIGBEE/stat/RESULT = {"Module":{"0":"Zigbee"}}

20:00:00.593 CMD: gpio 255
20:00:00.595 SRC: WebConsole from 172.30.30.100
20:00:00.596 CMD: Group 0, Index 1, Command "GPIO", Data "255"
20:00:00.603 MQT: home/TASMOTA_ZIGBEE/stat/RESULT = {"GPIO0":{"320":"Led_i1"},"GPIO1":{"3552":"Zigbee Tx"},"GPIO2":{"0":"None"},"GPIO3":{"3584":"Zigbee Rx"},"GPIO4":{"5312":"Zigbee Rst1"},"GPIO5":{"0":"None"},"GPIO9":{"0":"None"},"GPIO10":{"0":"None"},"GPIO12":{"640":"I2C SDA"},"GPIO13":{"544":"LedLink"},"GPIO14":{"608":"I2C SCL"},"GPIO15":{"0":"None"},"GPIO16":{"32":"Button1"},"GPIO17":{"0":"None"}}


```
- [ ] If using rules, provide the output of this command: `Backlog Rule1; Rule2; Rule3`:
```lua
  Rules output here:


```
- [x] Provide the output of this command: `Status 0`:
```lua
  STATUS 0 output here:

20:26:15.589 CMD: status 0
20:26:15.591 SRC: WebConsole from 172.30.30.100
20:26:15.593 CMD: Group 0, Index 1, Command "STATUS", Data "0"
20:26:15.597 MQT: home/TASMOTA_ZIGBEE/stat/STATUS = {"Status":{"Module":0,"DeviceName":"Tasmota","FriendlyName":["Tasmota"],"Topic":"TASMOTA_ZIGBEE","ButtonTopic":"0","Power":0,"PowerOnState":3,"LedState":0,"LedMask":"FFFF","SaveData":1,"SaveState":1,"SwitchTopic":"0","SwitchMode":[0,0,0,0,0,0,0,0],"ButtonRetain":1,"SwitchRetain":0,"SensorRetain":0,"PowerRetain":1,"InfoRetain":1,"StateRetain":1}}
20:26:15.602 MQT: home/TASMOTA_ZIGBEE/stat/STATUS1 = {"StatusPRM":{"Baudrate":115200,"SerialConfig":"8N1","GroupTopic":"tasmotas","OtaUrl":"http://ota.tasmota.com/tasmota/tasmota-zbbridge.bin","RestartReason":"Software/System restart","Uptime":"0T00:34:48","StartupUTC":"2021-05-18T18:51:27","Sleep":50,"CfgHolder":4617,"BootCount":251,"BCResetTime":"2021-02-21T12:36:05","SaveCount":420,"SaveAddress":"FD000"}}
20:26:15.608 MQT: home/TASMOTA_ZIGBEE/stat/STATUS2 = {"StatusFWR":{"Version":"9.4.0.4(zbbridge)","BuildDateTime":"2021-05-17T13:46:13","Boot":7,"Core":"2_7_4_9","SDK":"2.2.2-dev(38a443e)","CpuFrequency":160,"Hardware":"ESP8266EX","CR":"430/699"}}
20:26:15.611 MQT: home/TASMOTA_ZIGBEE/stat/STATUS3 = {"StatusLOG":{"SerialLog":0,"WebLog":3,"MqttLog":0,"SysLog":2,"LogHost":"jeroensteenhuis.nl","LogPort":514,"SSId":["xxxxx",""],"TelePeriod":60,"Resolution":"558180C0","SetOption":["00008069","2805C8000100060000005A00000000000000","00000080","00006080","00000180"]}}
20:26:15.618 MQT: home/TASMOTA_ZIGBEE/stat/STATUS4 = {"StatusMEM":{"ProgramSize":585,"Free":1204,"Heap":24,"ProgramFlashSize":2048,"FlashSize":2048,"FlashChipId":"1540A1","FlashFrequency":40,"FlashMode":3,"Features":["00000809","0F1007C4","04400001","00000002","00000000","00000000","00020000","00000000","00000000"],"Drivers":"1,2,7,9,10,20,23,41","Sensors":""}}
20:26:15.624 MQT: home/TASMOTA_ZIGBEE/stat/STATUS5 = {"StatusNET":{"Hostname":"TASMOTA_ZIGBEE","IPAddress":"172.30.31.54","Gateway":"172.30.30.254","Subnetmask":"255.255.254.0","DNSServer":"172.30.30.3","Mac":"84:CC:A8:9B:A2:74","Webserver":2,"WifiConfig":4,"WifiPower":17.0}}
20:26:15.630 MQT: home/TASMOTA_ZIGBEE/stat/STATUS6 = {"StatusMQT":{"MqttHost":"xxxxx","MqttPort":1883,"MqttClientMask":"DVES_%06X","MqttClient":"DVES_9BA274","MqttUser":"xxxxx","MqttCount":1,"MAX_PACKET_SIZE":1200,"KEEPALIVE":30,"SOCKET_TIMEOUT":4}}
20:26:15.636 MQT: home/TASMOTA_ZIGBEE/stat/STATUS7 = {"StatusTIM":{"UTC":"2021-05-18T19:26:15","Local":"2021-05-18T20:26:15","StartDST":"2021-03-28T02:00:00","EndDST":"2021-10-31T03:00:00","Timezone":"+01:00","Sunrise":"05:04","Sunset":"20:28"}}
20:26:15.639 MQT: home/TASMOTA_ZIGBEE/stat/STATUS10 = {"StatusSNS":{"Time":"2021-05-18T20:26:15"}}
20:26:15.646 MQT: home/TASMOTA_ZIGBEE/stat/STATUS11 = {"StatusSTS":{"Time":"2021-05-18T20:26:15","Uptime":"0T00:34:48","UptimeSec":2088,"Vcc":3.438,"Heap":24,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":1,"Wifi":{"AP":1,"SSId":"xxxxx","BSSId":"0E:EC:DA:8D:48:02","Channel":5,"RSSI":100,"Signal":-45,"LinkCount":1,"Downtime":"0T00:00:04"}}}


```
- [ ] Set `weblog` to 4 and then, when you experience your issue, provide the output of the Console log:
```lua
  Console output here:


```

### TO REPRODUCE
_Steps to reproduce the behavior:_

send CMD STATUS 0 with MQTT to Tasmota and see result.

This is the current result:
home/TASMOTA_ZIGBEE/stat/STATUS6 = {"StatusMQT":{"MqttHost":"xxxxxxx","MqttPort":1883,"MqttClientMask":"DVES_%06X","MqttClient":"DVES_9BA274","MqttUser":"jeroen","MqttCount":1,"MAX_PACKET_SIZE":1200,"KEEPALIVE":30,"SOCKET_TIMEOUT":4}}

### EXPECTED BEHAVIOUR
_A clear and concise description of what you expected to happen._

STATUS6 should show the ssl/tls state on mqtt

I would expect:
home/TASMOTA_ZIGBEE/stat/STATUS6 = {"StatusMQT":{"MqttHost":"xxxxx","MqttPort":1883,**"MqttSSL":0,**"MqttClientMask":"DVES_%06X","MqttClient":"DVES_9BA274","MqttUser":"jeroen","MqttCount":1,"MAX_PACKET_SIZE":1200,"KEEPALIVE":30,"SOCKET_TIMEOUT":4}}

### SCREENSHOTS
_If applicable, add screenshots to help explain your problem._

home/TASMOTA_ZIGBEE/stat/STATUS6 = {"StatusMQT":{"MqttHost":"xxxxxx","MqttPort":1883,"MqttClientMask":"DVES_%06X","MqttClient":"DVES_9BA274","MqttUser":"jeroen","MqttCount":1,"MAX_PACKET_SIZE":1200,"KEEPALIVE":30,"SOCKET_TIMEOUT":4}}

### ADDITIONAL CONTEXT
_Add any other context about the problem here._

**(Please, remember to close the issue when the problem has been addressed)**

**Comment by ascillato2** [comment]
> TO REPRODUCE
> Steps to reproduce the behavior:
> send CMD STATUS 0 with MQTT to Tasmota and see result

Please, can you share what is the result you are obtaining and which is the result you expect?

**Comment by jeroenst** [comment]
Of course, 
This is the current result:
```lua
home/TASMOTA_ZIGBEE/stat/STATUS6 = {"StatusMQT":{"MqttHost":"xxxxxxx","MqttPort":1883,"MqttClientMask":"DVES_%06X","MqttClient":"DVES_9BA274","MqttUser":"jeroen","MqttCount":1,"MAX_PACKET_SIZE":1200,"KEEPALIVE":30,"SOCKET_TIMEOUT":4}}
```

I would expect:
```lua
home/TASMOTA_ZIGBEE/stat/STATUS6 = {"StatusMQT":{"MqttHost":"xxxxx","MqttPort":1883,**"MqttSSL":0,**"MqttClientMask":"DVES_%06X","MqttClient":"DVES_9BA274","MqttUser":"jeroen","MqttCount":1,"MAX_PACKET_SIZE":1200,"KEEPALIVE":30,"SOCKET_TIMEOUT":4}}
```

**Reply by sfromis** [reply]
In general, I'd not expect SetOption values to also appear separately in Status command output. To query that value, use `SetOption103`. Strictly speaking, you can see the Value of SetOption103 if you parse the "SetOption" bits in `Status 3`. That would be flag bit 0x00200000 in the 4th value. Off in your case.

Still, there are lots of settings not showing up in Status command outputs.

**Reply by jeroenst** [reply]
SSL is just as important as port and clientid and should show up in this status info imo.

**Reply by jeroenst** [reply]
Anyway, thanks for the tip, I managed to get the status from the option bits.
